optaplanner-core/src/main/java/org/optaplanner/core/impl/score/definition/AbstractScoreDefinition.java

```diff
@@ -30,6 +32,38 @@
 
     private final String[] levelLabels;
 
+    protected static int sanitize(int number) {
+        return number == 0 ? 1 : number;
+    }
+
+    protected static long sanitize(long number) {
+        return number == 0L ? 1L : number;
+    }
+
+    protected static double sanitize(double number) {
+        return number == 0d ? 1d : number;
+    }
+
+    protected static BigDecimal sanitize(BigDecimal number) {
+        return number.signum() == 0 ? BigDecimal.ONE : number;
+    }
+
+    protected static int divide(int dividend, int divisor) {
+        return (int) Math.floor(divide(dividend, (double) divisor));
+    }
+
+    protected static long divide(long dividend, long divisor) {
+        return (long) Math.floor(divide(dividend, (double) divisor));
+    }
+
+    protected static double divide(double dividend, double divisor) {
+        return dividend / divisor;
+    }
+
+    protected static BigDecimal divide(BigDecimal dividend, BigDecimal divisor) {
+        return dividend.divide(divisor, dividend.scale() - divisor.scale(), RoundingMode.FLOOR);
```

Comment:
My spider sense is tingling, expecting comments here.
See table "Preferred Scales for Results of Arithmetic Operations" in [BigDecimal Javadoc.](https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html)

Reply:
This is fine. Our long existing method `Score.divide(boolean)` has a javadoc that says "When rounding is needed, it should be floored (as defined by {@link Math#floor(double)})." This is consistent with that.